### PR TITLE
feat: add support for using nvim-dap as debug runtime

### PR DESCRIPTION
**Pull request by JohnsonLee-debug**
With this pull request, you could use [nvim-dap](https://github.com/mfussenegger/nvim-dap) as debug runtime with just a few configuration like below.
image
Then, you could invoke debugging with nvim-dap through codeLens.
image

### CODE REVIEW

**Review by fannheyward — CHANGES_REQUESTED** [review]
Thank you for your PR, please do not format the `commands.ts` whole file.

**Comment by JohnsonLee-debug** [comment]
Sorry for the mistake. I've fixed it.

File: src/commands.ts [hunk 1]
```diff
@@ -724,11 +732,11 @@ export function viewMir(ctx: Ctx): Cmd {
 
 export function interpretFunction(ctx: Ctx): Cmd {
   return async () => {
-    const { document, position } = await workspace.getCurrentState();
+    const {document, position} = await workspace.getCurrentState();
     if (!isRustDocument(document)) return;
```
Comment (fannheyward) [inline src/commands.ts:736]: formatting issue too

**Review by fannheyward — CHANGES_REQUESTED** [review]

File: src/commands.ts (hunk 1 above)
Comment (JohnsonLee-debug) [reply src/commands.ts:736]: I think it would be a good idea to add local coc-server format configuration.

**Comment by JohnsonLee-debug** [comment]
I've fix the formatting issue again.

**Review by fannheyward — APPROVED** [review]

**Pull request merged**